src/main/java/io/ebeaninternal/server/query/CQuery.java

```diff
@@ -317,6 +317,9 @@ private boolean prepareBindExecuteQueryWithOption(boolean forwardOnlyHint) throw
         return false;
       }
 
+      // Call to native method nanoTime() while locked causes an expensive context switch.
+      // startNano is used only in updateExectuionStatistics, which itself is called in various places.
+      // Is this a problem?
       startNano = System.nanoTime();
```

Comment:
Intellij presented me with this: doing a call on a System method forces a switch of context, which can be expensive. Is this a problem?

The call happen inside a synchronize statement, feel free to look at the file.